### PR TITLE
Fix issue 23333: Make DList range access @nogc

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Unless I'm missing something this looked like an easy win to me.

Haven't done the full test-suite yet, and I am super out of date with what the newest dev/testing methodology is. Let me know if I've missed anything~

### CODE REVIEW

File: std/container/dlist.d [hunk 1]
```diff
@@ -62,6 +62,13 @@ module std.container.dlist;
     rs.popFront();
     nl.remove(rs);
     assert(equal(nl[], [1]));
+
+    () @nogc {
+        auto rn = nl[];
+        auto el = nl.front;
```
Comment (AndrejMitrovic) [inline std/container/dlist.d:68]: Should probably also test `.back`

Comment (RazvanN7) [reply std/container/dlist.d:68]: I'm not sure that this test is even required. I mean, if the code compiles with `@nogc:` that means that the functions that were annotated have passed compilation. I would simply drop this test.

**Comment by RazvanN7** [comment]
@AndrejMitrovic can you delete the test so that we can merge this?

**Comment by AndrejMitrovic** [comment]
@RazvanN7 done.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**